### PR TITLE
Decouple game

### PR DESCRIPTION
**Pull request by dvirk**
Make a timer object that stores time information from game. Humanoid objects now have this instead of reference to game. Also make a _wall_group for Humanoid to avoid call to game.walls.

### CODE REVIEW

File: src/sprites.py [hunk 1]
```diff
@@ -147,7 +165,7 @@ def update(self) -> None:
             except StopIteration:
                 self.damaged = False
 
-        self.rot = (self.rot + self.rot_speed * self.game.dt) % 360
+        self.rot = (self.rot + int(self.rot_speed * self._timer.dt)) % 360
```
Comment (mick-warehime) [inline src/sprites.py]: why does this need to be an int?
pull out that inner calculation into a new variable
delta_rot

File: src/sprites.py [hunk 2]
```diff
@@ -166,21 +184,21 @@ class Mob(Humanoid):
     def __init__(self, game: Any, pos: Vector2) -> None:
 
         super(Mob, self).__init__(images.MOB_IMG, settings.MOB_HIT_RECT, pos,
-                                  settings.MOB_HEALTH)
+                                  settings.MOB_HEALTH, game)
```
Comment (mick-warehime) [inline src/sprites.py]: same thing - pass in the groups

File: src/sprites.py [hunk 3]
```diff
@@ -59,37 +59,55 @@ class Humanoid(GameObject):
...
+    def current_time(self) -> int:
+        return pg.time.get_ticks()
+
+
 class Player(Humanoid):
     def __init__(self, game: Any, pos: Vector2) -> None:
         super(Player, self).__init__(images.PLAYER_IMG,
                                      settings.PLAYER_HIT_RECT, pos,
-                                     settings.PLAYER_HEALTH)
+                                     settings.PLAYER_HEALTH, game)
```
[the remark this answers is not in the document]
Comment (dvirk) [reply src/sprites.py]: Done

File: src/sprites.py (hunk 1 above)
Comment (dvirk) [reply src/sprites.py]: Done. It needs to be an int to match the type of self.rot.

**Pull request merged**